core/src/main/java/org/apache/brooklyn/feed/http/HttpFeed.java

```diff
@@ -275,7 +294,23 @@ public HttpFeed() {
     protected HttpFeed(Builder builder) {
         setConfig(ONLY_IF_SERVICE_UP, builder.onlyIfServiceUp);
         Map<String,String> baseHeaders = ImmutableMap.copyOf(checkNotNull(builder.headers, "headers"));
-        
+
+        HttpExecutor httpExecutor;
+        if (builder.httpExecutor != null) {
+            httpExecutor = builder.httpExecutor;
+        } else {
+            HttpExecutorFactory httpExecutorFactory = null;
+            Collection<? extends Location> locations = Locations.getLocationsCheckingAncestors(builder.entity.getLocations(), builder.entity);
+            Maybe<Location> location =  Machines.findUniqueElement(locations, Location.class);
```

Comment:
All of the elements here are of type `Location`.  It's possible to have two locations per entity - one coming from `location: xx` property in YAML  (which is the description of how to provision a new machine) and another `SshMachineLocation` (which points to a specific VM with IPs created from the previous one). I believe only the former will have the extensions when coming from yaml (they will not be inherited). Looking for the first location to have the required extension should be good enough.